java/maven/src/org/netbeans/modules/maven/modelcache/MavenProjectCache.java

```diff
@@ -171,6 +176,35 @@ public static boolean unknownBuildParticipantObserved(MavenProject project) {
         return project.getContextValue(CONTEXT_PARTICIPANTS) != null;
     }
     
+    /**
+     * Extracts a timestamp from the MavenProject. Timestamps are placed
+     * as context values in {@link #loadOriginalMavenProjectInternal}.
+     * @param p project
+     * @return timestamp, -1 if p is null, -2 if unknown
+     */
+    public static long getLoadTimestamp(MavenProject p) {
+        if (p == null) {
+            return -1;
+        }
+        Object o = p.getContextValue(CONTEXT_LOAD_TIMESTAMP);
+        if (o instanceof Long) {
+            return (long)o;
+        } else {
+            return -2;
+        }
```

Comment:
nitpick. one of my most favorite jdk 17 features is:
```java
    Object val = p.getContextValue(CONTEXT_LOAD_TIMESTAMP);
    return val instanceof Long l ? l : -2;
```

Reply:
Yeah, I have to get used to pattern matching

Reply:
Addressed  in 4c091822130a